src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/ModellingGroupController.kt

```diff
@@ -105,12 +106,69 @@ open class ModellingGroupController(context: ControllerContext)
 
     // TODO: https://vimc.myjetbrains.com/youtrack/issue/VIMC-307
     // Use streams to speed up this process of sending large data
-    fun getCoverageDataAndMetadata(context: ActionContext, repo: ModellingGroupRepository): SplitData<ScenarioTouchstoneAndCoverageSets, CoverageRow>
+    fun getCoverageDataAndMetadata(context: ActionContext, repo: ModellingGroupRepository):
+            SplitData<ScenarioTouchstoneAndCoverageSets, CoverageRow>
     {
         val path = ResponsibilityPath(context)
-        val data = repo.getCoverageData(path.groupId, path.touchstoneId, path.scenarioId)
-        checkTouchstoneStatus(data.structuredMetadata.touchstone.status, path.touchstoneId, context)
-        return data
+        val splitData = repo.getCoverageData(path.groupId, path.touchstoneId, path.scenarioId)
+        checkTouchstoneStatus(splitData.structuredMetadata.touchstone.status, path.touchstoneId, context)
+
+        val format = context.queryParams("format")
+
+        val tableData = when (format)
+        {
+
+            "wide" -> getWideDatatable(splitData.tableData.data)
+            "long", null -> splitData.tableData
+            else -> throw BadRequest("Format '$format' not a valid csv format. Available formats are 'long' " +
+                    "and 'wide'.")
+        }
+
+        return SplitData(splitData.structuredMetadata, tableData)
+    }
+
+    private fun getWideDatatable(data: Iterable<LongCoverageRow>):
+            FlexibleDataTable<WideCoverageRow>
+    {
+        val groupedRows = data
+                .groupBy { hashSetOf(it.countryCode, it.setName,
+                        it.ageFirst, it.ageLast,
+                        it.vaccine, it.gaviSupport, it.activityType) }//
+
+        val rows = groupedRows.values
+                .map {
+                    mapWideCoverageRow(it)
+                }
+
+        // all the rows should have the same number of years, so we just look at the first row
+        val years = rows.first().coverageAndTargetPerYear.keys.toList()
+
+        return FlexibleDataTable.new(rows, years)
+    }
+
+    private fun mapWideCoverageRow(records: List<LongCoverageRow>)
+            : WideCoverageRow
+    {
+        // all records have same country, gender, age_from and age_to, so can look at first one for these
+        val reference = records.first()
+
+        val coverageAndTargetPerYear = records.associateBy(
+                { "${it.year}_coverage" },
+                { it.coverage }).plus(records.associateBy(
+                { "${it.year}_target" },
+                { it.coverage }))
```

Comment:
Maybe easier to read as:

```
val coverageAndTargetPerYear = records.associateBy({ "${it.year}_coverage" }, { it.coverage })
                             + records.associateBy({ "${it.year}_target" }, { it.coverage })
```

Reply:
Won't this give all the coverage columns first, then all the target columns? I think it might be easier for the modellers to use if they are interleaved? Although maybe check with Tini.

Reply:
Wait, you've used `it.coverage` for both. The second one should be `it.target`, right?

Reply:
oh yikes, good catch... clearly my unit tests are deficient! will add test and fix this

Reply:
and yeah it will, I just figured this shouldn't matter, but I'll run by Tini